csrc/runtime/fusion_kernel_runtime.cpp

```diff
@@ -515,7 +515,30 @@ void FusionKernelRuntime::compileFusionParallel(KernelArgumentHolder args) {
             ", got ",
             group_to_run->schedulerType());
         if (group_to_run->schedulerType() == SchedulerType::Communication) {
-          // TODO: Implement communication lowering
+          auto deviceid = Communicator::getInstance().deviceId();
+          NVF_ERROR(
+              group_to_run->exprs().size() == 1,
+              "Communication segments must contain only one Expr");
+          HostIrLower lower;
+          for (auto* expr : lower.lower(
+                   ir_cloner.clone(group_to_run->exprs().at(0)), deviceid)) {
+            // Allocate the recv buffers of communications
+            if (expr->isA<Communication>()) {
```

Comment:
Can we NVF_ERROR this? IIUC, given group_to_run is communication-only, we expect all lowered expressions to be `Communication`.